File: zad1_2.py
class Vector2D():
    def __init__(self, xCoord, yCoord):
        self.x=xCoord
        self.y=yCoord

    def add(self, vec):
        return Vector2D(self.x+vec.x, self.y+vec.y)
    
    def multiply(self, vec):
        return (float(self.x*vec.x) + float(self.y*vec.y))

    def __str__(self):
        return ("[" + str(self.x) + ", " + str(self.y) + "]")

# test data 

vec1=Vector2D(-3, 1.4)
vec2=Vector2D(2, 2.7)
vec3=vec1.add(vec2)

vec4=vec1.multiply(vec2)

print(vec3)
print(vec4)

File: Java/zad1_2/zad1_2.java
package POB_2019.Java.zad1_2;

public class zad1_2 {
    public static void main(String[] args){
        Vector2D vec1 = new Vector2D(5, 11);
        Vector2D vec2 = new Vector2D(-2, 4);
        Vector2D vec3 = vec1.add(vec2);
        Vector2D vec4 = vec2.substract(vec1);
        double multiplicationResult = vec2.multiply(vec1);
        System.out.println(vec3);
        System.out.println(vec4);
        System.out.println(multiplicationResult);
    }
}